.github/workflows/build.yaml

```diff
@@ -0,0 +1,27 @@
+name: 'Build'
+
+on:
+  push:
+    branches-ignore:
+      - main
+  pull_request:
+    branches-ignore:
+      - main
+
+jobs:
+  linting-and-unit-testing:
+    runs-on: ubuntu-latest
```

Comment:
It is good practice to pin down versions, that way you don't get surprises anyhow. For example, ubuntu-latest is currently 22.04.4 but a major or minor version change of this could conceivably break this pipeline. Though this is unlikely for many reasons so just an FYI for this